tests/typecheck-tests.ts

```diff
@@ -999,6 +999,24 @@ type CreateTuple<L, V = string> =
       });
     });
   });
+
+  (function describe(_ = 'err') {
+    (function it(_ = 'infers the error type narrowly when it is a string') {
+      type Expectation = Result<never, 'error'>
+
+      const result = err('error')
+
+      const assignableToCheck: Expectation = result;
+    });
+
+    (function it(_ = 'infers the error type widely when it is not a string') {
+      type Expectation = Result<never, { abc: number }>
+
+      const result = err({ abc: 123 })
+
+      const assignableToCheck: Expectation = result;
+    });
+  })
```

Comment:
Side note - prettier can't run on this file because it's an outdated version and there's an 'import type'.

When I did try to run prettier on it, a LOT of changes were made, probably to do with the semicolons.

Reply:
@m-shaka mentioned we may want to use `biome` instead of prettier, but perhaps if we just update prettier this issue goes away?